static/fetch.js

```diff
@@ -1,31 +1,179 @@
-        this.date = new Date(date*1000);
-        this.parent = parent;
-        this.printer_name = printer_name;
-        this.username = username;
-        this.element = this.createElement();
-        this.symbol = Symbol();
+    constructor(printer, job) {
+        this.printer = printer;
+        this.username = job.username;
+        this.id = job.id;
+        this.last_updated = new Date(job.last_updated*1000);
+        this.createElement();
+        this.update(job);
+    }
+    update(job) {
+        this.status = job.status;
+        this.last_updated = new Date(job.last_updated*1000);
+        let stat = PrintJob.PARSE(this.status);
+        this.entry.status.innerText = stat.MSG;
+        this.entry.status.setAttribute('status', stat.ATTR);
+    }
+    getTime() {
+        const pad = (str, len) => {
+            str = String(str);
+            while (str.length < len) str = '0' + str;
+            return str;
+        }
+        let minutes = pad(this.last_updated.getMinutes(), 2);
+        let hours = pad(this.last_updated.getHours(), 2);
+        let AM = 'AM';
+        if (hours >= 13) {
+            AM = 'PM';
+            hours-=12
+        }
+        if (hours === 0) hours = 12;
+        return `${hours}:${minutes} ${AM}`;
     }
     createElement(){
-        let wrapper = document.createElement('div');
-        wrapper.classList.add('printlist-user');
-        let name = document.createElement('div');
-        name.classList.add('printlist-user-handle');
-        name.innerText = this.username;
-        wrapper.appendChild(name);
-        return wrapper;
+        this.entry = {
+            wrapper: _element('div', 'printlist-entry'),
+            time: _element('div', 'time-entry', null, `${this.getTime()}`),
+            username: _element('div', 'username-entry', null, this.username),
+            status: _element('div', 'status-entry')
+        };
+        this.entry.wrapper.appendChild(this.entry.time);
+        this.entry.wrapper.appendChild(this.entry.status);        
+        this.entry.wrapper.appendChild(this.entry.username);
+    }
+    clean(time){
+        const error = () => {
+            if (time - this.last_updated.getTime() > PERSISTENCE_TIME_ERROR)
+                this.remove();
+        }
+        switch (this.status) {
+            case PrintJob.COMPLETED.CODE:
+                if (time - this.last_updated.getTime() > PERSISTENCE_TIME_COMPLETE)
+                    this.remove();
+                break;
+            case PrintJob.PENDING.CODE:
+                if (time - this.last_updated.getTime() > PERSISTENCE_TIME_PENDING)
+                    this.remove();
+                break;
+            case PrintJob.FILE_ERROR.CODE:
+                error();
+                break;
+            case PrintJob.QUOTA_LIMIT.CODE:
+                error();
+                break;
+            case PrintJob.JOB_ERROR.CODE:
+                error();
+                break;
```

Comment:
you can use fallthrough to reduce duplication here and avoid the weird inline definition of `error`:
```js

            case PrintJob.FILE_ERROR.CODE:
            case PrintJob.QUOTA_LIMIT.CODE:
            case PrintJob.JOB_ERROR.CODE:
                if (time - this.last_updated.getTime() > PERSISTENCE_TIME_ERROR)
                    this.remove();
                break;
```